api/pkg/apis/v1alpha1/vendors/solutions-vendor.go

```diff
@@ -81,31 +87,40 @@ func (c *SolutionsVendor) onSolutions(request v1alpha2.COARequest) v1alpha2.COAR
 	if !exist {
 		namespace = constants.DefaultScope
 	}
+
+	version := request.Parameters["__version"]
```

Comment:
Versioning logic is scattered everywhere. Again, we keep to following the principal of separation of concern.